### PR TITLE
EI-610 set active collection calendar period

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

**Comment by ruslanss** [comment]
@Stephek had a good idea for a database constraint to only allow 1 active calendar period record. do you think we can add it as part of this story?

**Comment by David-B-Read** [comment]
> @Stephek had a good idea for a database constraint to only allow 1 active calendar period record. do you think we can add it as part of this story?

The inner API enforces only one active collection calendar period

File: src/SFA.DAS.EmployerIncentives.Api/Controllers/CollectionCalendarController.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        private readonly ILogger<CollectionCalendarController> _logger;
+
+        public CollectionCalendarController(IMediator mediator, ILogger<CollectionCalendarController> logger)
+        {
+            _mediator = mediator;
+            _logger = logger;
+        }
+
+        [HttpPatch]
+        [Route("/collectionCalendar/period/active")]
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api/Controllers/CollectionCalendarController.cs:22]: Shouldn't this be /collectionPeriods and the patch is just the CollectionPeriod data to be updated

File: src/SFA.DAS.EmployerIncentives.Api/Models/ActivateCollectionCalendarPeriodRequest.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿
+namespace SFA.DAS.EmployerIncentives.Api.Models
+{
+    public class ActivateCollectionCalendarPeriodRequest
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Api/Models/ActivateCollectionCalendarPeriodRequest.cs:4]: I think this is just a CollectionPeriodRequest with Number, Year and IsActive.  It doesn't make sense to call it Activate... when the Active field could be false.

**Review by Stephek — CHANGES_REQUESTED** [review]

**Review by Stephek — DISMISSED** [review]

**Review by ruslanss — APPROVED** [review]

**Pull request merged**